### PR TITLE
[v1.2][ISSUE-306] Add support to run pii remove with pyspark 

### PR DESCRIPTION
**Pull request by chaojun-zhang**
### What changes were proposed in this pull request?

Add pyspark dataframe as input of pii remove, return pyspark dataframe as output

### Why are the changes needed?

Easydata Integration

### How was this patch tested?

UT

### CODE REVIEW

File: RecDP/pyrecdp/primitives/llmutils/pii_remove.py [hunk 1]
```diff
@@ -130,54 +79,15 @@ def pii_remove(args):
...
-    columns = ["content", "modified", "secrets", "has_secrets", "number_secrets"]
-    ds_pii = ds_pii.remove_columns(columns)
-    ds_pii = ds_pii.rename_column("new_content", "content")
-    logger.info(f"Dataset info after removing columns:\n{ds_pii}")
+    args = getArgs()
+    logger.info(f"** The job is running with the following arguments: **\n{args}\n **** ")
 
-    # save the final dataset
+    spark = SparkSession.builder.master("local[{}]".format(args.cpu_per_worker)).appName(
+        "PII Remove").getOrCreate()
```
Comment (xuechendi) [inline RecDP/pyrecdp/primitives/llmutils/pii_remove.py:87]: @chaojun-zhang , let's using recdp embedded way to start spark, instead using SparkSession here

File: RecDP/pyrecdp/primitives/llmutils/pii_remove.py [hunk 2]
```diff
@@ -1,38 +1,23 @@
...
-
 import argparse
-import random
-import json
 import logging
-from pprint import pformat
-from functools import partial
 
-from datasets.utils.logging import set_verbosity_info
-from datasets import load_dataset
```
Comment (xuechendi) [inline RecDP/pyrecdp/primitives/llmutils/pii_remove.py:15]: did you add 'datasets' in setup.py, if so, may need to remove it as well

File: RecDP/pyrecdp/primitives/llmutils/pii_remove.py [hunk 3]
```diff
@@ -1,38 +1,23 @@
...
+from pyspark.sql import SparkSession
+from pyspark.sql.dataframe import DataFrame, Row
 
-from . pii.pii_detection import scan_pii_batch
-from . pii.pii_redaction import redact_pii_batch, random_replacements
```
Comment (xuechendi) [inline RecDP/pyrecdp/primitives/llmutils/pii_remove.py:18]: I fixed an issue here for lacking __init__.py in pii folder, FYI

File: RecDP/tests/test_llmutils.py [hunk 4]
```diff
@@ -74,22 +74,14 @@ def test_text_to_jsonl(self):
...
-            path: str = "json"
-            data_files: str = "tests/data/llm_data/arxiv_sample_100.jsonl"
-            split: str = "train"
-            text_column: str = "text"
-            batch_size: int = 100
-            num_proc: int = 8
-            seed: int = 10
-            save_path: str = "./pii_remove"
-            save_format: str = "json"
+        spark = SparkSession.builder.master("local[2]").appName(
```
Comment (xuechendi) [inline RecDP/tests/test_llmutils.py:79]: please use pyrecdp embedded way to start spark

File: RecDP/pyrecdp/primitives/llmutils/pii_remove.py (hunk 1 above)
Comment (chaojun-zhang) [reply RecDP/pyrecdp/primitives/llmutils/pii_remove.py:87]: Is there any reason to use this way?

File: RecDP/tests/test_llmutils.py (hunk 4 above)
Comment (chaojun-zhang) [reply RecDP/tests/test_llmutils.py:79]: @xuechendi what are the reasons for using this? I think this Spark data processor is a bit complex if only use for unit test?

File: RecDP/pyrecdp/primitives/llmutils/pii_remove.py (hunk 2 above)
Comment (chaojun-zhang) [reply RecDP/pyrecdp/primitives/llmutils/pii_remove.py:15]: No, package dataset seems to be added by you.

File: RecDP/tests/test_llmutils.py (hunk 4 above)
Comment (xuechendi) [reply RecDP/tests/test_llmutils.py]: @chaojun-zhang, the goal is to align with all other provided components with same API to enable spark.
if we are seeing issues, we don't have multiple branches. Please update to use the other interface.

Meanwhile, SparkDataProcess provides other materialization options.

**Review by xuechendi — APPROVED** [review]
LGTM

**Pull request merged**